### ISSUE TITLE
fix API changes 1.1.2 for _post_process_request

### ISSUE THREAD

**Issue by ramok** [trac-hacks/trac-listmilestones#1]
In trac 1.2 I got this error
```
2016-12-26 21:35:01,555 Trac[formatter] ERROR: Macro ListMilestones(boss-q1-2016) failed:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/trac/wiki/formatter.py", line 802, in _macro_formatter
    return macro.ensure_inline(macro.process(args))
  File "/usr/lib/python2.7/dist-packages/trac/wiki/formatter.py", line 379, in process
    text = self.processor(text)
  File "/usr/lib/python2.7/dist-packages/trac/wiki/formatter.py", line 351, in _macro_processor
    text)
  File "/usr/lib/python2.7/dist-packages/listmilestones/api.py", line 35, in expand_macro
    template, data, content_type = dispatcher._post_process_request(req, template, data, content_type)
ValueError: too many values to unpack
```
I found API changes in trac there https://trac.edgewall.org/changeset/12527.

This patch fix for me:

```
--- a/api.py	2016-12-26 21:57:28.478528109 +0100
+++ b/api.py	2016-12-26 21:58:39.000000000 +0100
@@ -32,9 +32,9 @@
         content_type = 'text/html'
 
         dispatcher = RequestDispatcher(self.env)
-        template, data, content_type = dispatcher._post_process_request(req, template, data, content_type)
+        template, data, content_type, method = dispatcher._post_process_request(req, template, data, content_type)
 
-        return Markup(Chrome(self.env).render_template(formatter.req, template, data, content_type=content_type, fragment=True))
+        return Markup(Chrome(self.env).render_template(formatter.req, template, data, content_type=content_type, fragment=True, method=method))
 
     def get_templates_dirs(self):
         from pkg_resources import resource_filename

```

**Comment by mitar** [comment]
Thanks. Could you make a pull request for this? One which checks how many elements `dispatcher._post_process_request` returns and based on that passes `method` on or not?

**Comment by ramok** [comment]
I make [pull request](https://github.com/trac-hacks/trac-listmilestones/pull/2). Straight fix, I did't find way make it more elegant ;]

**Comment by mitar** [comment]
Fixed in #2.
